test/cql-pytest/conftest.py

```diff
@@ -36,15 +36,20 @@ def pytest_addoption(parser):
 # We use scope="session" so that all tests will reuse the same client object.
 @pytest.fixture(scope="session")
 def cql(request):
-    # Use the default superuser credentials, which work for both Scylla and Cassandra
-    with cql_session(request.config.getoption('host'),
-        request.config.getoption('port'),
-        request.config.getoption('ssl'),
-        username="cassandra",
-        password="cassandra"
-    ) as session:
-        yield session
-        session.shutdown()
+    try:
+        # Use the default superuser credentials, which work for both Scylla and Cassandra
+        with cql_session(request.config.getoption('host'),
+            request.config.getoption('port'),
+            request.config.getoption('ssl'),
+            username="cassandra",
+            password="cassandra"
+        ) as session:
+            yield session
+            session.shutdown()
+    except NoHostAvailable:
+        # We couldn't create a cql connection. Instead of reporting that
+        # each individual test failed, let's just exit immediately.
+        pytest.exit(f"Cannot connect to Scylla at --host={request.config.getoption('host')} --port={request.config.getoption('port')}")
```

Comment:
`pytest.exit(reason, returncode=pytest.ExitCode.INTERNAL_ERROR)`

https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-exit

https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.ExitCode.INTERNAL_ERROR

Reply:
Ok, I'll do that and send a new patch, although the textual output looks exactly the same, and the exit code being 3 instead of 2 seems to not be relevant to anyone.
By the way, it's not strictly an internal error - it's an external error ;-)